### ISSUE TITLE
Install instructions with respect to $HOME/.terraform.d/plugins/ or $HOME/.terraform.d/plugins-cache/

### ISSUE THREAD

**Issue by techdragon** [GoogleCloudPlatform/terraformer#439]
At the moment the install instructions dont mention that you have to create a throwaway terraform directory, put files referencing the provider you intend to use, then execute `terraform init`, all to just put some files in the plugin directory `$HOME/.terraform.d/plugins/`. 

I realise this is mostly the terraform developers fault for not properly solving https://github.com/hashicorp/terraform/issues/15801 but following the install instructions for using a package manager basically dumped me strait down a rabbit hole and nerd sniped me. To help protect future nerds from getting sniped and wasting an hour fruitlessly searching for a solution it would be good to document/warn users here.  

**TLDR**: It would be good to document this requirement, provide a workaround if possible, and perhaps explain that a proper solution is at the mercy of Hashicorp perhaps linking to the closed issue that didn't actually get solved. https://github.com/hashicorp/terraform/issues/15801 

P.S. If this comes across as snarky, I apologise. The way that the linked issue ticket was handled got under my skin after spending an hour hunting for solutions, so I've done my best to share evidence of my frustrations as the example why this should be fixed without using language that is too loaded, but its a fine art at times. 😅

Edit: After running into a bug unrelated to this, I went to look at installing from source to get the fixes and I just discovered the install from source documentation covers most of this issue 🤦‍♂ Its probably worth rearranging the docs a bit so that people reading the package manager docs cant miss it like I did.

**Comment by meshuga** [comment]
Thanks for raising the issue. Looking at the tool from developer’s perspective is a little bit different...
I agree, that’s a good idea to rearrange the documentation. Along with that it would be great to split the documentation of the supported resources from the rest of readme. In its current form, with everything in one big blob, it’s hard to find anything.

**Comment by GaramNick** [comment]
I've just installed Terraformer from releases (Linux) with the provided four line snippet. When running against AWS I receive the following message.

```
2020/04/08 14:38:27 open /home/user/.terraform.d/plugins/linux_amd64: no such file or directory
```

At first sight, this seems related to your remark on the install instructions. Now, I'd like to know what the subsequent steps are I need to take to make this running because the README doesn't make that clear to me.

**Comment by tvvignesh** [comment]
@GaramNick Yup. Even I faced the same issue with plugins. Had to install using init.tf to solve the problem. README might need to be updated.

**Comment by techdragon** [comment]
not stale

**Comment by AshootoshSharma** [comment]
Not stale!!

**Comment by black-snow** [comment]
Same here and still haven't figured out what I have to do:

```
$ terraformer google list
2021/05/17 14:58:42 unknown command "google" for ""
$ terraformer list google
2021/05/17 14:58:50 unknown command "list" for ""
$ terraformer google import --help
2021/05/17 14:59:31 unknown command "google" for ""
```

Do I need to have `terraform` installed? If so, this should probs be stated in the README.  
I installed `terraformer` via homebrew - what else do I have to do now? Do I have to put one of [these](https://releases.hashicorp.com/terraform-provider-google/) somewhere? Do I have to add a `versions.tf` and `terraform init`? (and why?)

Also, if `terraform` is a prerequisite the brew formula should have it as dependency :)

**Comment by techdragon** [comment]
There are a number of issues that have gone stale, regarding failure to correctly discover plugins, and the issue has been complicated by the addition of newer versions of terraform not using the `.terraform.d/plugins/` folder, in fact they can break if you use that folder wrong. The recommended plugin-cache location should be searched in addition to the currently searched paths. https://www.terraform.io/docs/cli/config/config-file.html#provider-plugin-cache 

- https://github.com/GoogleCloudPlatform/terraformer/issues/590
- https://github.com/GoogleCloudPlatform/terraformer/issues/592
- https://github.com/GoogleCloudPlatform/terraformer/pull/608 ( Still has people commenting on it with issues )

**Comment by maresb** [comment]
Here's my procedure for getting terraform to populate the plugins cache so that terraformer can find it:

```bash
echo 'plugin_cache_dir = "$HOME/.terraform.d/plugins"' >> ~/.terraformrc
mkdir -p ~/.terraform.d/plugins
```

Since terraform caches the plugins by default in the `.terraform` subdirectory of the project root, rename or delete this directory so that the plugins need to be reinstalled. Finally
```bash
terraform init
```

**Comment by loki-dv** [comment]
Not stale, still have this issue with tf 0.14 and terraformer v0.8.18

**Comment by TheDroker** [comment]
Definitely not stale, so bumping

**Comment by r614** [comment]
bumping yet again, this issue still persists

**Comment by EverettHuang** [comment]
This problem will still arise now
OS: Linux64
terraform: v1.2.3
terraformer: v0.8.21

**Comment by M322458** [comment]
Same issue on Windows. Terraform init did not fix.

**Comment by chris-the-wiz** [comment]
the issue is still there. 
the fix that worked for me was:
1. initiating a new terraform  project ('temp_project') for the cloud provider  i need with 'terraform init' (in my case google)
2. cp temp_project/.terraform/providers/registry.terraform.io/hashicorp/google/5.20.0/linux_amd64/*   ~/.terraform.d/plugins/linux_amd64/
3. chmod 644  ~/.terraform.d/plugins/linux_amd64/terraform-provider-google_v5.20.0_x5

**Comment by luisnquin-invio** [comment]
Not stale

**Comment by Ocramius** [comment]
I'm running `terraformer` to reverse-engineer some work done in an AWS environment, and this worked for me at first startup

```sh
mkdir force_terraform_to_download_plugins
cd force_terraform_to_download_plugins
echo 'provider "aws" {}' > main.tf
terraform init
mkdir -p ~/.terraform.d/plugins/linux_amd64/
cp .terraform/providers/registry.terraform.io/hashicorp/aws/*/linux_amd64/* ~/.terraform.d/plugins/linux_amd64/
cd ..
rm -rf force_terraform_to_download_plugins
terraformer import aws --resources="*"
```

**Comment by miko-nd** [comment]
Still a problem
